typing/env.ml

```diff
@@ -976,33 +1192,24 @@ let lbl_shadow _lbl1 _lbl2 = false
 
 let lookup_value =
   lookup (fun env -> env.values) (fun sc -> sc.comp_values)
-and lookup_all_constructors =
+let lookup_all_constructors =
   lookup_all_simple (fun env -> env.constrs) (fun sc -> sc.comp_constrs)
     cstr_shadow
-and lookup_all_labels =
+let lookup_all_labels =
   lookup_all_simple (fun env -> env.labels) (fun sc -> sc.comp_labels)
     lbl_shadow
-and lookup_type =
+let lookup_type =
   lookup (fun env -> env.types) (fun sc -> sc.comp_types)
-and lookup_modtype =
+let lookup_modtype =
   lookup (fun env -> env.modtypes) (fun sc -> sc.comp_modtypes)
-and lookup_class =
+let lookup_class =
   lookup (fun env -> env.classes) (fun sc -> sc.comp_classes)
-and lookup_cltype =
+let lookup_cltype =
   lookup (fun env -> env.cltypes) (fun sc -> sc.comp_cltypes)
 
 let update_value s f env =
-  try
-    let ((p, vd), slot) = Ident.find_name s env.values in
-    match p with
-    | Pident id ->
-        let vd2 = f vd in
-        {env with values = Ident.add id ((p, vd2), slot) env.values;
-                  summary = Env_value(env.summary, id, vd2)}
-    | _ ->
-        env
-  with Not_found ->
-    env
+  (* update summary?? *)
```

Comment:
Ok, I've pushed a more complete fix: now the "copy_types" operation is exposed by Env and represented as such in the "summary".

After discussing with @garrigue, it seems the current situation is not ideal anyway, since the type of all values should be copied in non-principal mode, including those accessed with a module qualifier (which is not the case now, yielding to a different behavior when a value is accessed through an "open" or not).   Cleaning that would be nice but is quite independent of this PR, so I prefer to keep the current behavior untouched, and it seems cleaner to make the "copy" operation supported directly by Env.